PhoneBook.Web/Classes/EditFormPage.cs

```diff
@@ -85,47 +74,47 @@ protected bool SaveObject()
       EnsurePostLoadInvoked();
       EnsureValidatableControlsInitialized();
 
-      bool isValid = DataSource.Validate();
-      Control firstInvalidControl = null;
+      var isValid = DataSource.Validate();
 
-      foreach (DataEditUserControl control in DataEditUserControls)
+      var firstInvalidControl = DataEditUserControls.FirstOrDefault(control => !control.Validate());
```

Comment:
This code intends to validating all controls, the new code will only validate until the first control with a validation error, leaving potential additional issues hidden until the first error has been fixed.